internal/backend/webtools/db.go

```diff
@@ -0,0 +1,66 @@
+package webtools
+
+import (
+	"context"
+	"time"
+
+	"gorm.io/gorm"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db"
+)
+
+type terminalDB struct{ db.DB }
+
+type Message struct {
+	ID        uint           `gorm:"primarykey" json:"id"`
+	CreatedAt time.Time      `json:"created_at"`
+	DeletedAt gorm.DeletedAt `gorm:"index" json:"deleted_at"`
+
+	Addr string `json:"addr"`
+	Text string `json:"text"`
+}
+
+func newDB(db db.DB) *terminalDB { return &terminalDB{db} }
+
+func (db *terminalDB) Setup(ctx context.Context) error {
+	return db.db(ctx).AutoMigrate(&Message{})
+}
+
+func (db *terminalDB) db(ctx context.Context) *gorm.DB {
+	return (*gorm.DB)(db.DB.GormDB()).WithContext(ctx)
+}
+
+func (db *terminalDB) GetMessages(ctx context.Context, addr string) ([]Message, error) {
+	var msgs []Message
+
+	if err := db.db(ctx).Where("addr = ?", addr).Find(&msgs).Error; err != nil {
+		return nil, err
+	}
+
+	return msgs, nil
+}
+
+func (db *terminalDB) AddMessage(ctx context.Context, addr, text string) (uint, error) {
+	r := Message{Addr: addr, Text: text}
```

Comment:
Validate `addr` and `text` (not empty, maybe other checks)

Reply:
added validation for empty address, nothing much else to validate...